### DISCUSSION TITLE
Bulk queuing of programs

### DISCUSSION THREAD

**Discussion by skovatch** [Ascoware/get-iplayer-automator#422] [Announcements]
- Add support for getting all programs from a web page. The main way to use this feature is to open your web browser to a page like https://www.bbc.co.uk/programmes/m0015fn5/episodes/player, which has all episodes for a show -- in this case,  Bridge of Lies. Then, click 'Use Current Webpage'. Get iPlayer Automator will retrieve all of the program IDs and add them to the download queue. It will then attempt to fill in the metadata for the entries.
- Note: due to a limitation in get_iplayer it will not filter out shows already in the download history. But this just means that if you have downloaded any of the shows already, those downloads will fail and it will move on to the next in the queue.

<hr /><em>This discussion was created from the release <a href='https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.24.3'>Bulk queuing of programs</a>.</em>

**Comment by drguiom** [comment]
do you have any plans to do this from the iPlayer website directly? some shows do not show up when searching (not sure why, a force cache update doesn't make a difference) so I have to resort to clicking each episode for each season. See example attached: 
![Screen Shot 2022-11-06 at 10 58 54 AM](https://user-images.githubusercontent.com/31193546/200166912-3b431e74-7d62-4bf9-8ab6-957d52a5e178.png)

**Reply by SOLOSAIR** [reply]
I think if you scroll down on these pages you get to a link labelled 'programme website' which is the 'home' of a programme - then click on the 'Episodes' tab to see 'Available Now'. Try downloading from there? This is the type of page that Scott showed as an example of where to download them all from. Tip:  Googling 'name of programme' with the words _**bbc**_ and _**episodes**_ usually brings up the same result: look for the link with 'Episode Guide' within it. 🙂

**Reply by drguiom** [reply]
Yes that works for the shows that have a 'programme website' link, a lot don’t though.

**Comment by skovatch** [comment]
I’m not sure where it shows in the UI but get_iplayer caches at most 14 days of shows, which explains why you can’t search for really old content.

The page shown has a single episode on it, so I suspect Get Current Webpage is detecting that from the URL and only looking up one show. I suppose I could try doing a recursive search and if it returns nothing just use the current page logic.

**Reply by drguiom** [reply]
agreed the page here has a couple of things happening, the first episode is the "play next" link but the rest are showing under. It would be useful to iterate through them. I could then click to season 2 and do the same.

**Comment by hugofalkman** [comment]
Edit: To clarify, I submitted the screenshot, not as advertising for Downie, but as a possible implementation also for GiA. 

The screenshot below shows Downie's generic implementation of the functionality, using the London Kills webpage example. You just select the downloads you want from each webpage. Of course you only get to choose from the links actually displayed on each page, so you have to iterate, choosing links to download from each series separately. 

Note that for us, who can't use the "Get Current Webpage" method, because of always keeping too many browser windows and tabs open, the functionality should also be invokable, by copying a webpage link to the programme id field of a newly created download queue entry (using the + sign), followed by hitting the Process PID button.  

<img width="799" alt="Screenshot 2022-11-06 at 18 25 29" src="https://user-images.githubusercontent.com/10969068/200185645-4760697a-123f-4a17-97dc-9e15bba853c7.png">

**Reply by drguiom** [reply]
how do you get the screenshot above?

**Reply by hugofalkman** [reply]
You always get it, that’s how Downie works.

**Reply by drguiom** [reply]
oh I had not come across Downie!
